packages/framework/ai-collab/src/explicit-strategy/promptGeneration.ts

```diff
@@ -0,0 +1,390 @@
+		if (typeof value === "object" && !Array.isArray(value) && value !== null) {
+			// TODO: SharedTree Team needs to either publish TreeNode as a class to use .instanceof() or a typeguard.
+			// Uncomment this assertion back once we have a typeguard ready.
+			// assert(isTreeNode(node), "Non-TreeNode value in tree.");
+			const objId =
+				// eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+				idGenerator.getId(value) ?? fail("ID of new node should have been assigned.");
+			assert(
+				!Object.prototype.hasOwnProperty.call(value, objectIdKey),
+				`Collision of object id property.`,
+			);
+			return {
+				[objectIdKey]: objId,
+				...value,
+			} as unknown;
+		}
+		return value as unknown;
+	});
+	return stringified;
+}
+
+/**
+ * Generates a prompt designed to make an LLM produce a plan to edit the SharedTree to accomplish a user-specified goal.
+ */
+export function getPlanningSystemPrompt<TSchema extends ImplicitFieldSchema>(
+	view: TreeView<TSchema>,
+	treeNode: TreeNode,
+	userPrompt: string,
+	systemRoleContext?: string,
+): string {
+	const isRootNode = Tree.parent(treeNode) === undefined;
+	const schema = isRootNode
+		? normalizeFieldSchema(view.schema)
+		: normalizeFieldSchema(Tree.schema(treeNode));
+
+	const promptFriendlySchema = getPromptFriendlyTreeSchema(getJsonSchema(schema.allowedTypes));
+	const role = `I'm an agent who makes plans for another agent to achieve a user-specified goal to update the state of an application.${
+		systemRoleContext === undefined
+			? ""
+			: `
+			The other agent follows this guidance: ${systemRoleContext}`
+	}`;
+
+	const systemPrompt = `
+	${role}
+	The application state tree is a JSON object with the following schema: ${promptFriendlySchema}
+	The current state is: ${JSON.stringify(treeNode)}.
+	The user requested that I accomplish the following goal:
+	"${userPrompt}"
+	I've made a plan to accomplish this goal by doing a sequence of edits to the tree.
+	Edits can include setting the root, inserting, modifying, removing, or moving elements in the tree.
+	Here is my plan:`;
+
+	return systemPrompt;
+}
+
+/**
+ * Generates the main prompt of this explicit strategy.
+ * This prompt is designed to give an LLM instructions on how it can modify a SharedTree using specific types of {@link TreeEdit}'s
+ * and provides with both a serialized version of the current state of the provided tree node as well as  the interfaces that compromise said tree nodes data.
+ */
+export function getEditingSystemPrompt<TSchema extends ImplicitFieldSchema>(
+	userPrompt: string,
+	idGenerator: IdGenerator,
+	view: TreeView<TSchema>,
+	treeNode: TreeNode,
+	log: EditLog,
+	appGuidance?: string,
+	plan?: string,
+): string {
+	const isRootNode = Tree.parent(treeNode) === undefined;
+	const schema = isRootNode
+		? normalizeFieldSchema(view.schema)
+		: normalizeFieldSchema(Tree.schema(treeNode));
+	const promptFriendlySchema = getPromptFriendlyTreeSchema(getJsonSchema(schema.allowedTypes));
```

Comment:
I believe you should be able to simplify this to
```suggestion
	const promptFriendlySchema = getPromptFriendlyTreeSchema(getJsonSchema(schema));
```